indico/modules/rb_new/client/js/common/bookings/OccurrenceActionsDropdown.jsx

```diff
@@ -0,0 +1,180 @@
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import ReactDom from 'react-dom';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Form as FinalForm, Field} from 'react-final-form';
+import {Button, Icon, Confirm, Dropdown, Form, Portal, TextArea} from 'semantic-ui-react';
+
+import {ReduxFormField, formatters, validators as v} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import * as bookingsActions from './actions';
+
+import './OccurrenceActionsDropdown.module.scss';
+
+
+class OccurrenceActionsDropdown extends React.Component {
+    static propTypes = {
+        booking: PropTypes.object.isRequired,
+        date: PropTypes.string.isRequired,
+        actions: PropTypes.exact({
+            changeBookingOccurrenceState: PropTypes.func.isRequired,
+            fetchBookingDetails: PropTypes.func.isRequired
+        }).isRequired,
+    };
+
+    state = {
+        activeConfirmation: null,
+        dropdownOpen: false,
+        top: 0,
+        left: 0,
+    };
+
+    hideConfirm = () => {
+        this.setState({activeConfirmation: null});
+    };
+
+    showConfirm = (type) => {
+        this.setState({activeConfirmation: type});
+    };
+
+    changeOccurrenceState = (action, data = {}) => {
+        const {date} = this.props;
+        const {booking: {id}, actions: {changeBookingOccurrenceState, fetchBookingDetails}} = this.props;
+        changeBookingOccurrenceState(id, date, action, data).then(() => {
+            fetchBookingDetails(id);
+        });
+    };
+
+    findPositioning = () => {
+        const positioning = ReactDom.findDOMNode(this).getBoundingClientRect();
+        return {
+            top: positioning.top,
+            left: positioning.left
+        };
+    };
+
+    onClickButton = () => {
+        const {dropdownOpen} = this.state;
+        const {top, left} = this.findPositioning();
+        if (dropdownOpen) {
+            this.setState({
+                dropdownOpen: false,
+                top,
+                left
+            });
+        } else {
+            this.setState({
+                dropdownOpen: true,
+                top,
+                left
+            });
+        }
+    };
+
+    renderRejectionForm = ({handleSubmit, hasValidationErrors, submitSucceeded, submitting, pristine}) => {
+        return (
+            <Form styleName="rejection-form" onSubmit={handleSubmit}>
+                <Field name="reason"
```

Comment:
Maybe it should be appropriate here to display a message similar to the one displayed for cancellations.
> Are you sure you want to reject this occurrence (date)?

Reply:
Do we really need a confirmation there? I'd say having to enter a reason is confirmation enough